util/timer_test.cc

```diff
@@ -176,106 +104,76 @@ TEST_F(TimerTest, SingleScheduleRepeatedlyTest) {
 }
 
 TEST_F(TimerTest, MultipleScheduleRepeatedlyTest) {
-  uint64_t time_counter = 0;
-  mock_env_->set_current_time(0);
+  const int kInitDelay1Sec = 0;
+  const int kInitDelay2Sec = 1;
+  const int kRepeatSec = 2;
+  const int kIterations = 5;
+
+  int mock_time_sec = 0;
+  mock_env_->set_current_time(mock_time_sec);
   Timer timer(mock_env_.get());
 
-  InstrumentedMutex mutex1;
-  InstrumentedCondVar test_cv1(&mutex1);
-  const int kIterations1 = 5;
   int count1 = 0;
-  timer.Add(
-      [&] {
-        InstrumentedMutexLock l(&mutex1);
-        count1++;
-        if (count1 >= kIterations1) {
-          test_cv1.SignalAll();
-        }
-      },
-      "fn_sch_test1", 0, 2 * kSecond);
+  timer.Add([&] { count1++; }, "fn_sch_test1", kInitDelay1Sec * kSecond,
+            kRepeatSec * kSecond);
 
-  InstrumentedMutex mutex2;
-  InstrumentedCondVar test_cv2(&mutex2);
-  const int kIterations2 = 5;
   int count2 = 0;
-  timer.Add(
-      [&] {
-        InstrumentedMutexLock l(&mutex2);
-        count2++;
-        if (count2 >= kIterations2) {
-          test_cv2.SignalAll();
-        }
-      },
-      "fn_sch_test2", 1 * kSecond, 2 * kSecond);
+  timer.Add([&] { count2++; }, "fn_sch_test2", kInitDelay2Sec * kSecond,
+            kRepeatSec * kSecond);
 
   ASSERT_TRUE(timer.Start());
 
+  ASSERT_EQ(0, count2);
   // Wait for execution to finish
-  {
-    InstrumentedMutexLock l(&mutex1);
-    while(count1 < kIterations1) {
-      time_counter += kSecond;
-      mock_env_->set_current_time(time_counter);
-      test_cv1.TimedWait(time_counter);
-    }
+  for (; count1 < kIterations; mock_time_sec++) {
+    timer.TEST_WaitForRun([&] { mock_env_->set_current_time(mock_time_sec); });
+    ASSERT_EQ((mock_time_sec + 2) / 2, count1);
```

Comment:
I believe the provenance of 2 in `/ 2` is `kRepeatSec`, so `/ kRepeatSec` would be more clear.

Reply:
+1